SAT - 186 : [ui] add loading screen with a message in on-boarding last step

thestoneroller opened the pull request:
This PR moves the loading logic from `AuthContext` to Next.js's `loading.tsx` to streamline the loading process. It also introduces a pulsating animation around the logo to enhance user feedback during loading.

### **Changes Made**

- **`AuthContext.tsx`**:
    - Removed the loader rendering logic and associated imports.
- **`loading.tsx`**:
    - Created a new component to handle the page load animation before rendering the dashboard after onboarding.
- **`Loader.tsx`**:
    - Removed
- **`Logo.tsx`**:
    - Added `LoadingLogo` which only contains the half moon (center) part of the march logo 
- **`tailwind.config.ts`**:
    - Added `pulsate` keyframe animations to create a scale-up effect around the logo.

---

### **Checklist**
- [x] Tested locally
- [x] Synced with the latest `main` branch

---

### **Video**

![2024-10-1209-14-21-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/fa4c0f66-a665-4a0e-a459-6d6944c4cb0e)


### **Linked Issues**
fixes #212

thestoneroller committed:
[ui] add loading screen with a message in on-boarding last step

- Removed the `Loader` component(icon) and loading logic from `AuthContext`.
- Introduced Next.js `loading.tsx` to handle loading animations.
- Added `pulsate` animation keyframes in `tailwind.config.js`.

thestoneroller committed:
Revert "[ui] add loading screen with a message in on-boarding last step"

This reverts commit 0029dbaeae962900289dbdfb767b27cabe77d680.

thestoneroller committed:
[ui] add loading screen with a message in on-boarding last step

- Removed the `Loader` component(icon) and loading logic from `AuthContext`.
- Introduced Next.js `loading.tsx` to handle loading animations.
- Added `pulsate` animation keyframes in `tailwind.config.js`.

thestoneroller committed:
[ui] add loading screen with a message in on-boarding last step

- Removed the `Loader` component(icon) and loading logic from `AuthContext`.
- Introduced Next.js `loading.tsx` to handle loading animations.
- Added `pulsate` animation keyframes in `tailwind.config.js`.

thestoneroller committed:
Merge branch 'marchhq:main' into main

madhav-relish commented on apps/frontend/src/app/loading.tsx:
```diff
@@ -0,0 +1,22 @@
+          </div>
+        </div>
+        <div className="flex flex-col justify-center items-center gap-5">
+          <p className="text-[#EDEDED]">sit tight, loading some javascripts</p>
```
We already have predefined colors in tailwind.config.js, you can use that rather than adding a hex code.
Incase if the colors change in the future, we just have to change it at one place rather than changing it one by one

thestoneroller committed:
Merge branch 'ui/loading-screen' of https://github.com/TanmayAdithya/march into ui/loading-screen

madhav-relish commented on apps/frontend/src/contexts/AuthContext.tsx:
```diff
@@ -103,14 +102,6 @@ export function AuthProvider({
-      </div>
-    )
-  }
-
```
You can add your loader here if you want

thestoneroller replied:
Yeah, for some reason the loading.tsx was causing the build to fail. I still have no clue why it was doing that but now I switched back to the previous loading logic

thestoneroller committed:
Revert "[ui] add loading screen with a message in on-boarding last step"

This reverts commit 1b5a8b1d7b21a1b8562e83ffcb9da336abc77054.

thestoneroller committed:
[ui] add loading screen with a message in on-boarding last step

thestoneroller committed:
Merge branch 'main' of https://github.com/marchhq/march

thestoneroller committed:
Merge branch 'marchhq:main' into main

thestoneroller committed:
Merge branch 'main' of https://github.com/TanmayAdithya/march

thestoneroller committed:
Merge branch 'main' into ui/loading-screen

thestoneroller committed:
refactor: update StackConnect component with loading state

- Added loading state with loader when "Continue" is clicked.
- Replaced `Link` with `button` for delayed navigation using `useRouter`.
- Cleaned up unused code
